doc/designs/ech-api.md

```diff
@@ -0,0 +1,595 @@
+Encrypted ClientHello (ECH) APIs
+================================
+
+There is an [OpenSSL fork](https://github.com/sftcd/openssl/tree/ECH-draft-13c)
+that has an implementation of Encrypted Client Hello (ECH) and these are design
+notes relating to the current APIs for that, and an analysis of how these
+differ from those currently in the boringssl library.
+
+The [plan](https://github.com/openssl/project/issues/659) is to incrementally 
```

Comment:
doc-nits says you have some trailing whitespace here, but I'm not sure I see it.